component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -312,7 +320,35 @@ private String getErrorPage(AuthenticationContext context) throws Authentication
                 log.debug("Default authentication endpoint context is used");
             }
         }
-        return errorPage;
+        try {
+            return buildURL(errorPage, SMSOTPConstants.ERROR_PAGE);
+        } catch (URLBuilderException e) {
+            throw new AuthenticationFailedException("Error building SMS OTP error page URL.", e);
+        }
+    }
+
+    /**
+     * Get the mobile number request page from authentication.xml file or use the request page from constant file.
+     *
+     * @param context the AuthenticationContext
+     * @return the mobile number request page
+     * @throws AuthenticationFailedException
+     */
+    private String getMobileNoReqPage(AuthenticationContext context) throws AuthenticationFailedException {
+
+        String mobileNoReqPage = SMSOTPUtils.getMobileNumberRequestPage(context);
+        if (StringUtils.isEmpty(mobileNoReqPage)) {
+            mobileNoReqPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL()
+                    .replace(SMSOTPConstants.LOGIN_PAGE, SMSOTPConstants.MOBILE_NO_REQ_PAGE);
```

Comment:
what is it that we are doing here?

Reply:
We are resolving the authentication endpoint URL from the ConfigurationFacade and replacing the `/login.do` part with the mobile page related context `/mobile.jsp`. 
This is implemented similar to other sms otp pages [1].

[1] https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/blob/master/component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java#L289